### PR TITLE
#patch Fix restriction on maps/collections nesting

### PR DESCRIPTION
**Pull request by jschuchart-spot**
# TL;DR
Removing the restriction on nesting depth for collections and maps

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Removing the restriction on nesting depth for collections and maps. This simplifies the matching expression for SdkLiteralTypes.
This reuses the logic for creating TypeTags for a given type (-name) on the fly to make `SdkLiteralTypes.of(...)` recursive.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala [hunk 1]
```diff
@@ -72,87 +73,17 @@ object SdkLiteralTypes {
         blobs(BlobType.DEFAULT).asInstanceOf[SdkLiteralType[T]]
       case t if t =:= typeOf[Binary] =>
         binary().asInstanceOf[SdkLiteralType[T]]
+
+      case t if t <:< typeOf[List[Any]] =>
```
Comment (jschuchart-spot) [inline flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala:77]: technically, `List[T]` is also a `Product`, so I'm not sure how this match worked before

File: flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala [hunk 2]
```diff
@@ -369,28 +301,6 @@ object SdkLiteralTypes {
         }
       }
 
-      def createTypeTag[U <: Product](tpe: Type): TypeTag[U] = {
```
Comment (jschuchart-spot) [inline flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala:372]: pulled out to further down, so it is accessible in `.of()`

File: flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala [hunk 3]
```diff
@@ -499,6 +409,31 @@ object SdkLiteralTypes {
...
     }
+
+  private def createTypeTag[U](
+      mirror: universe.Mirror,
+      tpe: Type
+  ): TypeTag[U] = {
+    val typSym = mirror.staticClass(tpe.typeSymbol.fullName)
+    // note: this uses internal API, otherwise we will need to depend on scala-compiler at runtime
+    val typeRef =
+      universe.internal.typeRef(NoPrefix, typSym, tpe.typeArgs)
```
Comment (jschuchart-spot) [inline flytekit-scala_2.13/src/main/scala/org/flyte/flytekitscala/SdkLiteralTypes.scala:420]: Had to add the `tpe.typeArgs` as the last argument instead of `List.empty` as it would otherwise create a type tag e.g. for `List` instead of `List[String]`, which messes up the recursion
